inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceTest.java

```diff
@@ -0,0 +1,264 @@
+    private Project project;
+    private KnowledgeBase kb;
+    private TestFixtures testFixtures;
+    private Reification reification;
+    
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Autowired
+    private TestEntityManager testEntityManager;
+
+    @ClassRule
+    public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+    
+    @BeforeClass
+    public static void setUpOnce() {
+        System.setProperty("org.eclipse.rdf4j.repository.debug", "true");
+    }
+    
+    public KnowledgeBaseServiceTest(Reification aReification) {
+        reification = aReification;
+    }
+
+    @Parameterized.Parameters(name = "Reification = {0}")
+    public static Collection<Object[]> data()
+    {
+        return Arrays.stream(Reification.values()).map(r -> new Object[] { r })
+            .collect(Collectors.toList());
+    }
+    
+    @Before
+    public void setUp() throws Exception {
+        EntityManager entityManager = testEntityManager.getEntityManager();
+        testFixtures = new TestFixtures(testEntityManager);
+        sut = new KnowledgeBaseServiceImpl(temporaryFolder.getRoot(), entityManager);
+        project = testFixtures.createProject(PROJECT_NAME);
+    }
+
+    @After
+    public void tearDown() throws Exception
+    {
+        testEntityManager.clear();
+        sut.destroy();
+    }
+    
+    @Test
+    public void testListPropertiesKnowledgeBaseIRI_WithLocalOWL() throws Exception
+    {
+        kb = buildKnowledgeBase(project, KB_NAME.concat("-" + "OWL"));
+        kb.setType(RepositoryType.LOCAL);
+        sut.registerKnowledgeBase(kb, sut.getNativeConfig());
+        setSchema(kb, OWL.CLASS, RDFS.SUBCLASSOF, RDF.TYPE, RDFS.COMMENT, RDFS.LABEL, RDF.PROPERTY);
+        importKnowledgeBase("data/wine-ontology.rdf");
+        List<KBHandle> propertiesKBHandle = sut.listProperties(kb,  RDF.PROPERTY, true, true);
+        log.debug("\n \n Size of List Properties OWL::::::::" + propertiesKBHandle.size() + "\n \n");
+        assertThat(propertiesKBHandle).as("Check that list is not empty").isNotEmpty();
+
+    }
+    
+    
+    @Test
+    public void testListPropertiesKnowledgeBaseIRI_WithWikidata() throws IOException
+    {
+        String kbProfile = "wikidata";
+        KnowledgeBaseProfile  profile = sut.readKnowledgeBaseProfiles().get(kbProfile);
+        KnowledgeBaseMapping mapping = profile.getMapping();
+        kb = buildKnowledgeBase(project, KB_NAME.concat("-" + kbProfile));
+        kb.setType(RepositoryType.REMOTE);
+        sut.registerKnowledgeBase(kb, sut.getRemoteConfig(profile.getSparqlUrl()));
+        setSchema(kb, mapping.getClassIri(), mapping.getSubclassIri(), mapping.getTypeIri(),mapping.getDescriptionIri(), mapping.getLabelIri(), mapping.getPropertyTypeIri());
```

Comment:
Please split these in 3 blocks, 1 for setup 1 for the action you test and 1 for assertions.

Reply:
I am removing the class, as it has been already changed to https://github.com/inception-project/inception/blob/473f67a64f1fef3fd47db7cca6e8a95d78ee2989/inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceRemoteTest.java#L70